dot/parachain/candidate-validation/worker.go

```diff
@@ -42,10 +67,33 @@ func (w *worker) executeRequest(task *workerTask) (*ValidationResult, error) {
 		logger.Debugf("candidate %x already processed", candidateHash)
 		return processed, nil
 	}
-	validationResult, err := w.instance.ValidateBlock(task.work)
-	if err != nil {
-		logger.Errorf("executing validate_block: %w", err)
-		reasonForInvalidity := ExecutionError
+
+	var validationResult *parachainruntime.ValidationResult
+	validationResultCh := make(chan (*resultWithError))
+	timeoutDuration := determineTimeout(task.timeoutKind)
+
+	go func() {
+		result, err := w.instance.ValidateBlock(task.work)
+		if err != nil {
+			validationResultCh <- &resultWithError{result: nil, err: err}
+		}
+		validationResultCh <- &resultWithError{
+			result: result,
+		}
```

Comment:
so, if there is an error you will end up sending result twice.

```suggestion
		if err != nil {
			validationResultCh <- &resultWithError{result: nil, err: err}
		} else{
		    validationResultCh <- &resultWithError{
			    result: result,
		    }
        }
```

Reply:
Good catch, updated.